go-controller/pkg/util/dpu_annotations.go

```diff
@@ -137,16 +151,30 @@ func UnmarshalPodDPUConnStatusAllNetworks(annotations map[string]string) (map[st
 	return podDcss, nil
 }
 
-// MarshalPodDPUConnStatus adds the pod's connection status of the specified network to the corresponding pod annotation.
-func MarshalPodDPUConnStatus(annotations map[string]string, scs *DPUConnectionStatus, netName string) (map[string]string, error) {
+// MarshalPodDPUConnStatus adds the pod's connection status of the specified NAD to the corresponding pod annotation.
+// if scs is nil, delete the pod's connection status of the specified NAD
+func MarshalPodDPUConnStatus(annotations map[string]string, scs *DPUConnectionStatus, nadName string) (map[string]string, error) {
 	if annotations == nil {
 		annotations = make(map[string]string)
 	}
 	podScss, err := UnmarshalPodDPUConnStatusAllNetworks(annotations)
 	if err != nil {
 		return nil, err
 	}
-	podScss[netName] = *scs
+	sc, ok := podScss[nadName]
+	if scs != nil {
+		if ok && sc == *scs {
+			return nil, newAnnotationAlreadySetError("OVN pod %s annotation for NAD %s already exists in %v",
+				DPUConnetionStatusAnnot, nadName, annotations)
+		}
+		podScss[nadName] = *scs
+	} else {
+		if !ok {
+			return nil, newAnnotationAlreadySetError("OVN pod %s annotation for NAD %s already removed",
+				DPUConnetionStatusAnnot, nadName)
+		}
+		delete(podScss, nadName)
```

Comment:
Same as the above comment about, when will we ever delete a Pod's DPU connection status.

Reply:
when connectionDetails is deleted, correspondingly, its dpu connectionStatus annotation needs to be deleted. see https://github.com/cathy-zhou/ovn-kubernetes/blob/dpuCNIFix/go-controller/pkg/node/base_node_network_controller_dpu.go#L161.

Reply:
Again, thanks for the explanation. It is clear now.